src/hypofuzz/dashboard/test.py

```diff
@@ -103,33 +104,34 @@ def _check_invariants(self) -> None:
         # assert self.elapsed_time == total_elapsed_time
 
     def add_report(self, report: Report) -> None:
-        # we use last_report to compute timestamp_monotonic, and last_report_worker
-        # to compute status_count_diff and elapsed_time_diff
-        last_report = self.linear_reports[-1] if self.linear_reports else None
-        last_worker_report = (
-            None
-            if report.worker_uuid not in self.reports_by_worker
-            else self.reports_by_worker[report.worker_uuid][-1]
-        )
-        if (
-            last_worker_report is not None
-            and last_worker_report.elapsed_time > report.elapsed_time
-        ):
-            # If last_report.elapsed_time > report.elapsed_time, the reports have arrived
-            # out of order. we've already accounted for the diff of `report`; last_report
-            # essentially became the combined report of report + last_report.
+        last_worker_report = None
+        reports_index = 0
+        if report.worker_uuid in self.reports_by_worker:
+            reports = self.reports_by_worker[report.worker_uuid]
+            # If a report is added out of order, then the appropriate report to
+            # compute the diff against is the one right before the new report.
+            # Any guaranteed-monotonic attribute will work here (either
+            # elapsed_time or status_counts). Use elapsed_time.
             #
-            # TODO instead of dropping this report, insert it into the lists at
-            # the appropriate index, and recompute the diff for the next
-            # report. That way the e.g. dashboard graph still gets the report.
-            return
+            # we expect reports to *usually* arrive in-order, which case the
+            # appropriate report to diff against is reports[-1].
+            reports_index = bisect_right(
+                reports, report.elapsed_time, key=lambda r: r.elapsed_time
+            )
```

Comment:
... I now think that we should probably define a `fast_bisect_right` helper which checks the end, then the start, and _then_ bisects if necessary.  log-n -> constant in the average case is pretty good!